doc/sphinx/proof-engine/ltac2.rst

```diff
@@ -890,10 +893,19 @@ Ltac2 from Ltac1
 Same as above by switching Ltac1 by Ltac2 and using the `ltac2` quotation
 instead.
 
-Note that the tactic expression is evaluated eagerly, if one wants to use it as
-an argument to a Ltac1 function, one has to resort to the good old
-:n:`idtac; ltac2:(foo)` trick. For instance, the code below will fail immediately
-and won't print anything.
+.. productionlist:: coq
+  ltac_expr : ltac2 : ( `ltac2_term` )
+            : ltac2 : ( `ident` ... `ident` |- `ltac2_term` )
+
+The typing rules are dual, that is, the optional identifiers are bound
+with type `Ltac2.Ltac1.t` in the Ltac2 expression, which is expected to have
+type unit. The value returned by this quotation is an Ltac1 function with the
+same arity as the number of bound variables.
+
+Note that when no variables are bound, the inner tactic expression is evaluated
+eagerly, if one wants to use it as an argument to a Ltac1 function, one has to
+resort to the good old :n:`idtac; ltac2:(foo)` trick. For instance, the code
```

Comment:
I suggest the following improvement, related to my question in #10908:

```suggestion
resort to the good old :n:`ltac:( idtac; ltac2:(@ltac2_expr) )` trick. For instance, the code
```

Reply:
This doesn't scale insofar as you might actually build more complex Ltac1 expressions involving the quotation. This is written for people already knowing the quirks of Ltac1, otherwise what's the point of hooking into a broken language?

Reply:
OK, as you wish.